File: Python/2444.py
class Solution:
    def countSubarrays(self, nums: list[int], minK: int, maxK: int) -> int:
        ans = 0
        cur_min = -1
        cur_max = -1
        invalid = -1

        for r in range(len(nums)):
            if nums[r] == minK:
                cur_min = r
            if nums[r] == maxK:
                cur_max = r
            if nums[r] < minK or nums[r] > maxK:
                invalid = r
            
            ans += max(0, min(cur_max, cur_min)-invalid)

        return ans

File: C++/2444.cpp
#include <algorithm>
#include <vector>
class Solution {
public:
    long long countSubarrays(std::vector<int>& nums, int minK, int maxK) {
        long long ans = 0;
        int cur_min = -1, cur_max = -1, invalid = -1;
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] == minK){
                cur_min = i;
            }
            if(nums[i] == maxK){
                cur_max = i;
            }
            if(nums[i] < minK || nums[i] > maxK){
                invalid = i;
            }
            ans += std::max(0, (std::min(cur_min, cur_max)-invalid));
        }
        return ans;
    }
};